core/domain/user_jobs_one_off.py

```diff
@@ -323,6 +326,78 @@ def reduce(user_id, stringified_commit_times_msec):
+    """Job that checks the html content of an exploration Draft change List and
+     validates all the Math tags in the HTML.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [user_models.ExplorationUserDataModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function for this job."""
+        draft_change_list = item.draft_change_list
+        html_list = []
+        for change in draft_change_list:
+            if not change['cmd'] == exp_domain.CMD_EDIT_STATE_PROPERTY:
+                continue
+            new_value = change['new_value']
+            if change['property_name'] == exp_domain.STATE_PROPERTY_CONTENT:
+                html_list = html_list + [new_value['html']]
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_CUST_ARGS):
+                # Only customization args with the key 'choices' have HTML
+                # content in them.
+                if 'choices' in new_value.keys():
+                    html_list = html_list + new_value['choices']['value']
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_WRITTEN_TRANSLATIONS):
+                written_translations_object = (
+                    state_domain.WrittenTranslations.from_dict(new_value))
+                html_list = (
+                    html_list +
+                    written_translations_object.get_all_html_content_strings())
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_DEFAULT_OUTCOME):
+                html_list = html_list + [new_value['feedback']['html']]
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_HINTS):
+                for hint in new_value:
+                    html_list = html_list + [hint['hint_content']['html']]
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_SOLUTION):
+                html_list = html_list + [new_value['explanation']['html']]
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_ANSWER_GROUPS):
+                for answer_group in new_value:
+                    answer_group_object = (
+                        state_domain.AnswerGroup.from_dict(answer_group))
+                    html_list = (
+                        html_list +
+                        answer_group_object.get_all_html_content_strings())
+
+        html_string = ''.join(html_list)
+        error_list = (
+            html_validation_service.validate_math_tags_in_html(html_string))
+        if len(error_list) > 0:
+            yield ({
+                'draft_id': item.id,
+                'exp_id': item.exploration_id,
+                'user_id': item.user_id,
```

Comment:
is adding the `last_updated_time` to the output helpful here?

Reply:
Not really

Reply:
Assume someone has very old drafts for which we don't have the conversion methods and we expect them to be discarded eventually so do we want to audit those draft change list? if yes, then do we want to fix them (considering we don't have a cob=version method for them I don't think we can)? if now, then I think we should have some info to indicate that the draft change list belongs to a very old exp/state schema.

Reply:
Oh I see, in which case should you then not be outputting the exp/state schema instead? But sure, if you want to output the last updated time to it, I don't have an objection, it isn't sensitive data and can give us some idea of the recency of the changes.

Reply:
I've added the last_updated field too.

> in which case should you then not be outputting the exp/state schema instead?

Do you mean the state schema version number?

Reply:
Sorry, missed this. Yes, I meant the state schema version number.